tests/cpp/plugin/test_federated_learner.cc

```diff
@@ -13,66 +13,119 @@
-  std::shared_ptr<DMatrix> sliced{Xy_->SliceCol(world_size, rank)};
-  std::unique_ptr<Learner> learner{Learner::Create({sliced})};
-  learner->SetParam("tree_method", "approx");
-  learner->SetParam("objective", "binary:logistic");
-  learner->UpdateOneIter(0, sliced);
   Json model{Object{}};
   learner->SaveModel(&model);
   ASSERT_EQ(model, expected_model);
 }
 
-TEST_F(FederatedLearnerTest, BaseScore) {
-  std::shared_ptr<DMatrix> Xy_{RandomDataGenerator{kRows, kCols, 0}.GenerateDMatrix(true)};
-  std::unique_ptr<Learner> learner{Learner::Create({Xy_})};
-  learner->SetParam("tree_method", "approx");
-  learner->SetParam("objective", "binary:logistic");
-  learner->UpdateOneIter(0, Xy_);
-  Json config{Object{}};
-  learner->SaveConfig(&config);
-  auto base_score = GetBaseScore(config);
-  ASSERT_NE(base_score, ObjFunction::DefaultBaseScore());
+class FederatedLearnerTest : public BaseFederatedTest {
+ protected:
+  static auto constexpr kRows{16};
+  static auto constexpr kCols{16};
 
-  RunWithFederatedCommunicator(kWorldSize, server_address_, &VerifyBaseScore, kRows, kCols,
-                               base_score);
-}
+  void TestObjective(std::string const& objective) {
+    std::shared_ptr<DMatrix> dmat{RandomDataGenerator{kRows, kCols, 0}.GenerateDMatrix(true)};
 
-TEST_F(FederatedLearnerTest, Model) {
-  std::shared_ptr<DMatrix> Xy_{RandomDataGenerator{kRows, kCols, 0}.GenerateDMatrix(true)};
-  std::unique_ptr<Learner> learner{Learner::Create({Xy_})};
-  learner->SetParam("tree_method", "approx");
-  learner->SetParam("objective", "binary:logistic");
-  learner->UpdateOneIter(0, Xy_);
-  Json model{Object{}};
-  learner->SaveModel(&model);
+    auto &h_upper = dmat->Info().labels_upper_bound_.HostVector();
+    auto &h_lower = dmat->Info().labels_lower_bound_.HostVector();
+    h_lower.resize(kRows);
+    h_upper.resize(kRows);
+    for (size_t i = 0; i < kRows; ++i) {
+      h_lower[i] = 1;
+      h_upper[i] = 10;
+    }
 
-  RunWithFederatedCommunicator(kWorldSize, server_address_, &VerifyModel, kRows, kCols,
-                               std::cref(model));
-}
+    std::unique_ptr<Learner> learner{Learner::Create({dmat})};
+    learner->SetParam("tree_method", "approx");
+    learner->SetParam("objective", objective);
+    if (objective.find("quantile") != std::string::npos) {
+      learner->SetParam("quantile_alpha", "0.5");
+    }
+    if (objective.find("multi") != std::string::npos) {
+      learner->SetParam("num_class", "3");
+    }
+    learner->UpdateOneIter(0, dmat);
+    Json config{Object{}};
+    learner->SaveConfig(&config);
+    auto base_score = GetBaseScore(config);
+
+    Json model{Object{}};
+    learner->SaveModel(&model);
+
+    RunWithFederatedCommunicator(kWorldSize, server_address_, &VerifyBaseScoreAndModel, kRows,
+                                 kCols, objective, base_score, model);
+  }
+};
+
+TEST_F(FederatedLearnerTest, RegSquaredError) { TestObjective("reg:squarederror"); }
```

Comment:
I think this https://github.com/dmlc/xgboost/blob/4caca2947df22fa76af27bbaad379aafe9f6c51d/tests/cpp/objective/test_objective.cc#L30 would be more future-proofed.

Reply:
Done.